src/main/java/com/commercetools/sync/services/impl/TaxCategoryServiceImpl.java

```diff
@@ -49,8 +50,9 @@ public TaxCategoryServiceImpl(@Nonnull final ProductSyncOptions syncOptions) {
                 if (StringUtils.isNotBlank(fetchedTaxCategoryKey)) {
                     keyToIdCache.put(fetchedTaxCategoryKey, id);
                 } else {
-                    syncOptions.applyWarningCallback(format("TaxCategory with id: '%s' has no key set. Keys are"
-                        + " required for taxCategory matching.", id));
+                    syncOptions.applyWarningCallback(
+                        new SyncException(format("TaxCategory with id: '%s' has no key set. Keys are"
+                        + " required for taxCategory matching.", id)), null, null);
```

Comment:
Because we don't have it here. It fetches list of `TaxCategory` and it is using `ProductSyncOptions`. Types don't match (`TaxCategory` is not `Product`).